app/src/test/groovy/pl/futurecollars/invoicing/controllers/tax/TaxCalculatorControllerTest.groovy

```diff
@@ -0,0 +1,89 @@
+package pl.futurecollars.invoicing.controllers.tax
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.http.MediaType
+import org.springframework.test.web.servlet.MockMvc
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers
+import pl.futurecollars.invoicing.service.InvoiceService
+import pl.futurecollars.invoicing.service.JsonService
+import pl.futurecollars.invoicing.service.TaxCalculatorService
+import spock.lang.Specification
+import static pl.futurecollars.invoicing.TestHelper.company
+import static pl.futurecollars.invoicing.TestHelper.invoice
+
+@SpringBootTest
+@AutoConfigureMockMvc
+class TaxCalculatorControllerTest extends Specification {
+
+    @Autowired
+    private MockMvc mockMvc
+
+    @Autowired
+    private JsonService jsonService
+
+    @Autowired
+    private TaxCalculatorService taxCalculatorService
+
+    @Autowired
+    private InvoiceService invoiceService
+
+    def setup() {
+        invoiceService.getAll().each { invoice -> invoiceService.delete(invoice.id) }
+    }
+
+    def "should get response 404 - not found when get nonexistent taxIdentificationNumber from id 1"() {
+        expect:
```

Comment:
Nazwa testu - not found when no value is given

Reply:
poprawione